### ISSUE TITLE
Muti-output Support?

### ISSUE THREAD

**Issue by yupbank** [dmlc/xgboost#2033]
Hi, i did help a bit with sklearn multi-output module, and xgboost have really good performance in both accuracy and speed.
currently i'm using it by  
``` python
X, Y = load_data()
for i in xrange(Y.shape[1]):
    dtrain = xgb.DMatrix(X, Y[:,i])
    bst = xgb.train( plst, dtrain, num_round, evallist )
    bst.save_model('%s.model'%i)
```
which is really slow, even for DMatrix support two dimension Y is already better..
